feature/subscriber-attributes/src/main/java/com/revenuecat/purchases/subscriberattributes/caching/SubscriberAttributesCache.kt

```diff
@@ -57,21 +57,25 @@ class SubscriberAttributesCache(
         getAllStoredSubscriberAttributes(appUserID).filterUnsynced(appUserID)
 
     @Synchronized
-    fun clearSubscriberAttributesIfSyncedForSubscriber(appUserID: String) {
-        val unsyncedSubscriberAttributes = getUnsyncedSubscriberAttributes(appUserID)
-        if (unsyncedSubscriberAttributes.isNotEmpty()) {
-            return
-        }
+    fun clearAllSubscriberAttributesFromUser(appUserID: AppUserID) {
         log(LogIntent.DEBUG, AttributionStrings.DELETING_ATTRIBUTES.format(appUserID))
         val allStoredSubscriberAttributes = getAllStoredSubscriberAttributes()
         val updatedStoredSubscriberAttributes =
             allStoredSubscriberAttributes.toMutableMap().also {
                 it.remove(appUserID)
             }.toMap()
-
         deviceCache.putAttributes(updatedStoredSubscriberAttributes)
     }
 
+    @Synchronized
+    fun clearSubscriberAttributesIfSyncedForSubscriber(appUserID: AppUserID) {
+        val unsyncedSubscriberAttributes = getUnsyncedSubscriberAttributes(appUserID)
+        if (unsyncedSubscriberAttributes.isNotEmpty()) {
+            return
+        }
+        clearAllSubscriberAttributesFromUser(appUserID)
```

Comment:
I thinks this would be easier to follow like:
```suggestion
        if (unsyncedSubscriberAttributes.isEmpty()) {
            clearAllSubscriberAttributesFromUser(appUserID)
        }
```